hw/ip/prim/dv/prim_present/crypto_dpi_present/present.inc

```diff
@@ -0,0 +1,392 @@
+/*
+ * Implementation of PRESENT in C
+ * v2.1, 10/13/2008
+ *
+ * Thomas Siebert, thomas.siebert@rub.de
+ *
+ *
+ * Your Compiler currently should support
+ * the ANSI-C99-standard.
+ *
+ * Tested with gcc (with Option -std=c99)
+ */
+
+//----------------------------------
+// Includes
+//----------------------------------
+#include "boxes.inc"    //S-Boxes and P-Boxes
+#include "comline.inc"  //Command Line
+#include "verbose.inc"  //For verbose output
+#include <getopt.h>
+#include <inttypes.h>
+#include <stdint.h>
+#include <stdio.h>  //Standard C headers...
+#include <stdlib.h>
+
+//----------------------------------
+// Macros for bit manipulation
+//----------------------------------
+//returns...
+#define high45_64(h45in) ((uint64_t)h45in >> 9)            // 45 msb as lsb
+#define high61_64(h4in) ((uint64_t)h4in >> 3)              // 61 msb as lsb
+#define high4_64(h4in) ((uint64_t)h4in >> 60)              // 4 msb as lsb
+#define high8to4_64(h8in) (((uint64_t)h8in >> 56) & 0x0F)  // 4 msb as 2. lsb
+#define high16_64(h16in) ((uint64_t)h16in >> 48)           // 16 msb as lsb
+#define high1_64(h1in) ((uint64_t)h1in >> 63)              // msb as lsb
+#define low4_64(l4in) ((uint64_t)l4in << 60)               // 4 lsb as msb
+#define low8to4_64(l4in) ((uint64_t)l4in << 56)            // 4 lsb as 2. msb
+#define low16_64(l4in) ((uint64_t)l4in << 48)              // 4 lsb as msb
+#define rotate1l_64(r1lin) \
+  (high1_64(r1lin) | (r1lin << 1))  // input rotated left (1x)
+#define rotate1r_64(r1rin) \
+  (high1_64(r1rin) | (r1rin >> 1))  // input rotated right (1x)
+#define rotate4l_64(r4lin) \
+  (high4_64(r4lin) | (r4lin << 4))  // input rotated left (4x)
+#define rotate4r_64(r4rin) \
+  (high4_64(r4rin) | (r4rin >> 4))  // input rotated right (4x)
+
+//----------------------------------
+// Function prototypes
+//----------------------------------
+uint64_t encrypt(uint64_t, uint64_t *, uint16_t, _Bool);
+uint64_t decrypt(uint64_t, uint64_t *, uint16_t, _Bool);
+uint64_t *key_schedule(uint64_t, uint64_t, uint16_t, _Bool, _Bool);
+
+//----------------------------------
+// Start of code
+//----------------------------------
+// int main( int argc, char ** const argv )
```

Comment:
can you add a comment here explaining why this section has been commented out?

Reply:
Sure, will do.